test/deep.test.js

```diff
@@ -144,6 +144,67 @@ describe('--deep with nested ncurc files', function () {
     deepJsonOut['pkg/sub2/package.json'].should.not.have.property('cute-animals')
     deepJsonOut['pkg/sub2/package.json'].should.have.property('fp-and-or')
     deepJsonOut['pkg/sub2/package.json'].should.have.property('ncu-test-v2')
+
+    // pkg3: reject: ['cute-animals']
+    deepJsonOut.should.have.property('pkg/sub3/package.json')
+    deepJsonOut['pkg/sub3/package.json'].should.not.have.property('cute-animals')
+    deepJsonOut['pkg/sub3/package.json'].should.have.property('fp-and-or')
+    deepJsonOut['pkg/sub3/package.json'].should.have.property('ncu-test-v2')
+  })
+
+  it('use ncurc of nested packages with --recursive option', async () => {
+
+    const configFilePath = path.join(__dirname, 'deep-ncurc', '.ncurc.js')
+    const deepJsonOut = await spawn('node', [bin, '--jsonUpgraded', '--deep', '--recursive', '--configFilePath', configFilePath], { cwd }).then(JSON.parse)
```

Comment:
If you specify `--configFilePath` on the command line, it will use that config file on every nested package anyway (without `--recursive`), as command line options apply to every package. If you omit this option, it will use the config file in `cwd`.

Reply:
Without `--configFilePath` tests not works as implementation have a side-effects depended on `process.env.NCU_TESTS`
and internal state is diffrent due:
```javascript
// load .ncurc
// Do not load when global option is set
// Do not load when tests are running (an be overridden if configFilePath is set explicitly)
const rcResult = !program.global && (!process.env.NCU_TESTS || configFilePath)
  ? ncu.getNcurc({ configFileName, configFilePath, packageFile })
  : null
```
Behevior is very implicit and not obvious. Same code rinning from cli and from test has a different behevior.
As a lot of tests depends on `process.env.NCU_TESTS` it was a most simple way to arcjive the predictable behevior.

Reply:
> Behevior is very implicit and not obvious. Same code rinning from cli and from test has a different behevior.
> As a lot of tests depends on `process.env.NCU_TESTS` it was a most simple way to achieve the predictable behevior without core code modifications.

I agree, not the best solution. The tests should probably always run in a clean directory.

I like your solution of checking `recursive` to ensure the config file gets loaded.